block_course_contents.php

```diff
@@ -164,15 +164,41 @@ public function get_content() {
                 $title = $format->get_section_name($section);
             }
 
+            if ( ($i == 0) && (!empty($globalconfig->display_course_link)) ) {
```

Comment:
We should not check the globalconfig here. The global config should be used for setting the default values. Each block instance should have these settings available (via edit_form.php file).